kedro/io/data_catalog.py

```diff
@@ -160,6 +162,12 @@ def __init__(  # noqa: PLR0913
             >>>                   save_args={"index": False})
             >>> catalog = DataCatalog(datasets={'cars': cars})
         """
+        warnings.warn(
+            "`DataCatalog` has been deprecated and will be replaced by an improved alternative, `KedroDataCatalog`, in Kedro 1.0.0."
+            "After this change, the `DataCatalog` name will persist, but its functionality will align with `KedroDataCatalog`. "
+            "For more details, refer to the documentation: https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature",
```

Comment:
```suggestion
            "`DataCatalog` has been deprecated and will be replaced by `KedroDataCatalog` in Kedro 1.0.0."
            "After this change, the `DataCatalog` name will persist, but its functionality will align with `KedroDataCatalog`. "
            "For more details, refer to the documentation: https://docs.kedro.org/en/stable/data/index.html#kedrodatacatalog-experimental-feature",
```

While I agree it's an improvement it's up to the user to decide if it is 😉 this way the message is shorter as well.